src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -758,18 +762,50 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
         SHOW_GIVER_NAME: lastQuestionEditFormModel.showGiverNameTo,
         SHOW_RECIPIENT_NAME: lastQuestionEditFormModel.showRecipientNameTo,
       });
-      this.newQuestionEditFormModel.showResponsesTo =
+      const newQuestionShowResponsesTo: FeedbackVisibilityType[]  =
           newQuestionVisibilityStateMachine.getVisibilityTypesUnderVisibilityControl(VisibilityControl.SHOW_RESPONSE);
-      this.newQuestionEditFormModel.showGiverNameTo =
+      const newQuestionShowGiverNameTo: FeedbackVisibilityType[] =
           newQuestionVisibilityStateMachine.getVisibilityTypesUnderVisibilityControl(VisibilityControl.SHOW_GIVER_NAME);
-      this.newQuestionEditFormModel.showRecipientNameTo =
-          newQuestionVisibilityStateMachine.getVisibilityTypesUnderVisibilityControl(
-              VisibilityControl.SHOW_RECIPIENT_NAME);
+      const newQuestionShowRecipientNameTo: FeedbackVisibilityType[] =
+          newQuestionVisibilityStateMachine
+              .getVisibilityTypesUnderVisibilityControl(VisibilityControl.SHOW_RECIPIENT_NAME);
+
+      let isAllowedToUseInheritedVisibility: boolean = false;
+      if (this.feedbackQuestionsService
+          .isCustomFeedbackVisibilitySettingAllowed(this.newQuestionEditFormModel.questionType)) {
+        isAllowedToUseInheritedVisibility = true;
```

Comment:
The original issue was only for team contribution questions but it seems this will prevent any inheritance for questions without custom visibility settings allowed? Or are team contribution questions the only with that setting?

Reply:
In this case: team contribution questions the only with that setting.

In the future case where we add new question that only allows common feedback visibility setting, the same logic should also follow I believe (or we will end up with a new issue raised).